metadata-ingestion/src/datahub/ingestion/source/sql/stored_procedures/base.py

```diff
@@ -0,0 +1,246 @@
+    language: str
+    extra_properties: Optional[Dict[str, str]]
+
+    def get_procedure_identifier(
+        self,
+    ) -> str:
+        argument_signature_suffix = (
+            ProcedureSignature(argument_signature=self.argument_signature).guid()
+            if self.argument_signature
+            else ""
+        )
+        return f"{self.name}_{argument_signature_suffix}"
+
+    def to_urn(self, database_key: DatabaseKey, schema_key: Optional[SchemaKey]) -> str:
+        return make_data_job_urn(
+            orchestrator=database_key.platform,
+            flow_id=get_procedure_flow_name(database_key, schema_key),
+            job_id=self.get_procedure_identifier(),
+            cluster=database_key.env or DEFAULT_ENV,
+            platform_instance=database_key.instance,
+        )
+
+
+def generate_flow_workunits(
+    database_key: DatabaseKey, schema_key: Optional[SchemaKey]
+) -> Iterable[MetadataWorkUnit]:
+    """Generate flow workunits for database and schema"""
+
+    procedure_flow_name = get_procedure_flow_name(database_key, schema_key)
+
+    flow_urn = make_data_flow_urn(
+        orchestrator=database_key.platform,
+        flow_id=procedure_flow_name,
+        cluster=database_key.env or DEFAULT_ENV,
+        platform_instance=database_key.instance,
+    )
+
+    yield MetadataChangeProposalWrapper(
+        entityUrn=flow_urn,
+        aspect=DataFlowInfoClass(
+            name=procedure_flow_name,
+        ),
+    ).as_workunit()
+
+    yield MetadataChangeProposalWrapper(
+        entityUrn=flow_urn,
+        aspect=SubTypesClass(
+            typeNames=[FlowContainerSubTypes.MSSQL_PROCEDURE_CONTAINER],
+        ),
+    ).as_workunit()
+
+    if database_key.instance:
+        yield MetadataChangeProposalWrapper(
+            entityUrn=flow_urn,
+            aspect=DataPlatformInstanceClass(
+                platform=make_data_platform_urn(database_key.platform),
+                instance=make_dataplatform_instance_urn(
+                    platform=database_key.platform,
+                    instance=database_key.instance,
+                ),
+            ),
+        ).as_workunit()
+
+    yield MetadataChangeProposalWrapper(
+        entityUrn=flow_urn,
+        aspect=ContainerClass(container=database_key.as_urn()),
+    ).as_workunit()
+
+
+def get_procedure_flow_name(
+    database_key: DatabaseKey, schema_key: Optional[SchemaKey]
+) -> str:
+    if schema_key:
+        procedure_flow_name = (
+            f"{schema_key.database}.{schema_key.db_schema}.stored_procedures"
+        )
+    else:
+        procedure_flow_name = f"{database_key.database}.stored_procedures"
+    return procedure_flow_name
+
+
+def generate_job_workunits(
+    database_key: DatabaseKey,
+    schema_key: Optional[SchemaKey],
+    procedure: BaseProcedure,
+) -> Iterable[MetadataWorkUnit]:
+    """Generate job workunits for database, schema and procedure"""
+
+    # TODO: escape special chars if present in stored procedure name
```

Comment:
we recently fixed this concern for dataflow/datajobs https://github.com/datahub-project/datahub/pull/12772 have you found it again?

Reply:
No, Its something that mssql stored procs do, so had added here as TODO. Guessing its not needed then.